Add sig-net description of sig-chair role (#7034)
kubernetes/community: sig-network/role-sig-chair.md

# Role: SIG Chair

## Background

Please see the
[standard description of a SIG chair](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#chair),
which does a good job describing the expectations of the more “administrative”
aspect of being a SIG chair.

> # What’s expected

Reviewer: ```suggestion
## What's expected
```

Rather than repeat the doc linked above, this doc will just add some color.

The minimum effective engagement for a SIG chair is to “run the show” (really
“make sure the show is running” - delegation is good!).  That generally means
building an agenda for and running regular meetings, curating our
community-facing metadata, and acting as liaison to other groups (other SIGs
and steering, mostly).

This includes new-issue triage - either doing the triage or leading it (e.g. in
the regular SIG meetings).

> # Above and beyond

Reviewer: ```suggestion
## Above and beyond
```

The basic expectations above should not be a huge time commitment.  SIG chairs
who want to have more impact and can commit more time can engage in a multitude
of additional ways.  While this can include being an active code contributor or

> SIG TL, an effective SIG chair does not NEED to be a code contributor.

Reviewer: ```suggestion
SIG TL, an effective SIG chair does not **need** to be a code contributor.
```
We can use Markdown emphasis.

Some concrete things that SIGs need to do, which the chair-people can help with

> include:

Reviewer 2: Commenting as this was brought up as a possible thing we could adopt across all SIGs at the Jan. 2023 Leads meetings.

Reading through this doc initially, I didn't catch that the bulleted list that follows is intended as the responsibilities of the *SIG*, since the doc is geared towards the chair. (thanks @marosset for that catch!) My initial reaction was that this list of expectations for the SIG chair is simply far too much for a single person to be expected to do.

I would suggest that for clarity, this doc be reframed as "these are the things that need to happen in the SIG" rather than as expectations of the chairs.

It may also be relevant to share the leads curriculum I wrote back in 1.22, which includes areas of responsibility for SIG Leads as well as homework to better understand the various parts of SIGs: https://github.com/kubernetes/community/blob/master/mentoring/programs/leads-curriculum.md

Reviewer: _optional_ change

To avoid confusing readers:
```suggestion
There's a longer list of concrete things that SIGs need to do, and which the chair-people can
help. As a chair, these are **not** your responsibility, but the chair is accountable for
taking steps to help make things happen.

Examples of these activities include:
```
?
Does that help?

Author: The intention was really "these are things that most/all SIGs do (or should), any or all which an engaged sig-chair can facilitate or even do themselves.

Reviewer 2: I just don't think it's realistic for an individual SIG Chair to be attempting to do all of the above by themselves. It feels a little bit like writing aspirational responsibilities for a job and then being surprised that people don't apply because they can't imagine themselves doing more than 50% of it.

By putting this in a "SIG Chair" doc rather than "SIG Activities" doc, this comes across to me as a list of activities we expect an individual to be doing rather than the group of SIG volunteers, even though we have the brief caveat written out. I don't think that framing is sustainable, and so in its current form I wouldn't support rolling this out as requirements for all chairs.

I would much rather see us writing documentation that encourages us to build SIGs where responsibilities and recognition are both widely shared amongst a large pool of volunteers vs. setting up more single points of failure and "heroes."

Author: Not "all".

The previous text said "can help with" and "can engage in a multitude of additional ways".

I have revised it to say "Some concrete things that SIGs need to do, which the chair-people can facilitate, participate in, or even do themselves include:"

I don't think I am asking for or implying heroics, am I?  I'm just trying to say "here's a bunch of things you can do it you want to engage more"

Reviewer 3: @ehashman I think many of these activities are already a part of the chair definition. This writeup just expands on those. What I feel happens here is that this document talks about bare minimum chair responsibilities when everything that CAN be delegated IS delegated. In situations when it's true, not delegating some of these items is expanded in above and beyond section.

Knowing that not everything will be delegated, seeing this list may be scary. Suggestion to change it to "SIG Activities" may be a reflection of the fact that delegation today is not trivial with lack of time and energy in SIGs. Calling these responsibilities "SIG Activities"  shifts the focus from "chair have to actively delegate/do things" -> "common list of tasks that everybody have to contribute to". It also helps to highlight the fact that all of the above MUST happen.

I believe that actively delegating is an expectation from a chair. So if we are keeping this document structure, I'd suggest we highlight this as a primary responsibility. 

I also suggested more changes here: https://github.com/kubernetes/community/pull/7034#issuecomment-1378459260

    updates, etc.
  * Manage issue assignments - go through issues which are assigned to people
    and see if progress is being made or if assignments are stale.
  * Curate "help wanted" and "good first issue" issues - file new issues or
    improve existing issues (e.g. by adding details or getting others to do so)
    to enable new contributors to take them up.
  * Manage project board(s) to organize information - things like KEPs, issues,
    PRs, or in-progress dev efforts can usually benefit from organizing and
    curation.
  * Manage KEPs - track the KEP lifecycle, update metadata, ping KEP owners for
    updates, etc.
  * Organize events - new member introductions, KubeCon meetups, local events,
    etc.
  * Actively solicit topics for regular meetings - reach out to KEP owners or
    other efforts which could benefit from SIG discussion, or which other
    attendees would benefit from hearing about.
  * Docs - create, curate, edit, revise, or restructure docs (API docs, website
    docs, etc.) related to the SIG.
  * Check in and monitor subprojects - keep an eye out for stale / potential
    new OWNERS to elevate or move to emeritus.
  * ...get creative!

> Many of these things will need help from major technical contributors or even the SIG TL(s).  SIG chairs are, first and foremost, facilitators, and the SIG TLs are their partners.

Reviewer: ```suggestion
Many of these things will need help from major technical contributors or even the SIG technical lead(s)
(also known as TLs).  SIG chairs are, first and foremost, facilitators, and the SIG TLs are their partners.
```

# Role: SIG Chair

## Background

Please see the
[standard description of a SIG chair](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#chair),
which does a good job describing the expectations of the more “administrative”
aspect of being a SIG chair.

## What's expected

Rather than repeat the doc linked above, this doc will just add some color.

The minimum effective engagement for a SIG chair is to “run the show” (really
“make sure the show is running” - delegation is good!).  That generally means
building an agenda for and running regular meetings, curating our
community-facing metadata, and acting as liaison to other groups (other SIGs

> and steering, mostly).

Reviewer 3: there is also community updates, annual report, enhancement tracking, etc. Is this paragraph summarizing the link https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#chair or trying to boil it down to "_important_ bare minimum"?

Reviewer 3: maybe the what's expected section can be changed to exclude this section and concentrate on the importance of other things like new issues triage and trends analysis? For big SIGs this is an important addition that is not obvious from the description in the link

Author: I wanted to emphasize that there is a "basic" role.  It may play down this aspect, I'll add a few words, burt I don't want to repeat too much.

> This includes new-issue triage - either doing the triage or leading it (e.g. in

Reviewer 3: issues and PRs? Or just issues?

I think this also needs to include tracking the trends of open/closed issues.

Reviewer 3: trends are important to watch to make sure the list is being worked on. Bugs can be triaged, but nobody is working on them

the regular SIG meetings).

## Above and beyond

> The basic expectations above should not be a huge time commitment.  SIG chairs

Reviewer 3: > The basic expectations above should not be a huge time commitment.

are "basic expectations" refers to ones outlined in this doc or in the linked requirements document?  Neither should be a huge time commitment (with the common sense definition of "huge"). The sentence is, however, sounds discrediting for the work that can be done.

Author: If one does the minimum expectations, it really shouldn't be a ton of work.  I don't mean that dismissively or to imply there's not value to it.  It's just not going to eat 10 hours a week.  Or it shouldn't, at least.

SIG tech-lead, an effective SIG chair does not **need** to be a code
contributor.

Some concrete things that SIGs need to do, which the chair-people can
facilitate, participate in, or even do themselves include:
  * Active new-issue triage - go through new issues, ask for more info if
    needed, apply labels, close or redirect “support” requests, CC people who
    might have context, de-dup, triage-accept obvious bugs, ping submitters for
    updates, etc.
  * Backlog grooming - go through older issues, re-evaluate triage, de-dup,
    aggregate similar issues, cross-link issues, link to KEPs, probe for
    updates, etc.
  * Manage issue assignments - go through issues which are assigned to people
    and see if progress is being made or if assignments are stale.
  * Curate "help wanted" and "good first issue" issues - file new issues or
    improve existing issues (e.g. by adding details or getting others to do so)
    to enable new contributors to take them up.
  * Manage project board(s) to organize information - things like KEPs, issues,
    PRs, or in-progress dev efforts can usually benefit from organizing and
    curation.

>   * Manage KEPs - track the KEP lifecycle, update metadata, ping KEP owners for

Reviewer 3: this is a requirement from the [document](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#chair:~:text=SHOULD%20identify%2C%20track%2C%20and%20maintain%20the%20SIGs%20enhancements%20for%20current%20release%20and%20serve%20as%20point%20of%20contact%20for%20the%20release%20team%2C%20but%20MAY%20delegate%20to%20another%20Lead%20to%20fulfill%20these%20responsibilities):

> SHOULD identify, track, and maintain the SIGs enhancements for current release and serve as point of contact for the release team, but MAY delegate to another Lead to fulfill these responsibilities

Does this refers to the fact that to go beyond one should not delegate? Or that tracking may be more involved than just checking the status of PRs in enhancements repo?

Author: At least in sig-net, much of the KEP tracking is done as a group.  I think the basic requirement is pinging people, but there's a lot more that CAN be done.  Will update a bit.
